### PR TITLE
superlu 5.2.2

### PR DESCRIPTION
**Pull request by carlocab**
Created manually without `brew bump-formula-pr`. :(

- bump version to 5.2.2
- switch build to CMake (https://github.com/Homebrew/homebrew-core/pull/63276)
- add license (https://github.com/xiaoyeli/superlu/blob/a8a0caafa593772450b35339c8fa4e76b7bc934f/License.txt)

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/superlu.rb [hunk 1]
```diff
@@ -19,25 +19,21 @@ class Superlu < Formula
...
-            "SUPERLULIB=$(SuperLUroot)/lib/libsuperlu.a",
-            "CC=#{ENV.cc}",
-            "BLASLIB=-L#{Formula["openblas"].opt_lib} -lopenblas"]
-
-    system "make", "lib", *args
-    cd "EXAMPLE" do
-      system "make", *args
+    args = std_cmake_args + %W[
+      -Denable_internal_blaslib=NO
+      -DTPL_BLAS_LIBRARIES=#{Formula["openblas"].opt_lib}/libopenblas.dylib
```
Comment (iMichka) [inline Formula/superlu.rb:29]: Can you use the `shared_libraries` DSL here, to make this work on linux too?

**Review by iMichka — DISMISSED** [review]
One small comment above but looks good

File: Formula/superlu.rb (hunk 1 above)
Comment (carlocab) [reply Formula/superlu.rb:29]: Sure, just not 100% sure how that works. 

Does that just involve `s/libopenblas.dylib/#{shared_library("libopenblas")}`?

Comment (iMichka) [reply Formula/superlu.rb:29]: Exactly. This will expand to the right extension based on the platform.

Comment (carlocab) [reply Formula/superlu.rb:29]: `git push -f` is taking ages, but the commits should be on their way soon.

**Review by fxcoudert — APPROVED** [review]

**Pull request closed without merging**